### ISSUE TITLE
Failure to take and persist `createAccessIntent`  based SAF permissions

### ISSUE THREAD

**Issue by d4rken** [d4rken/sdmaid-public#1496]
```
1517587076075 I/Debug: F: VERSIONNAME:4.9.5; VERSIONCODE:40905
1517587076075 I/Debug: P: VERSIONNAME: -- ; VERSIONCODE: -- 
1517587076075 D/Debug: Fingerprint: Nokia/TA-1004_00WW/NB1:8.0.0/OPR6.170623.013/00WW_4_39A:user/release-keys
1517587076075 D/Debug: ro.build.version.codename=REL
1517587076075 D/Debug: ro.build.version.incremental=00WW_4_39A
1517587076076 D/Debug: ro.build.version.base_os=Nokia/TA-1004_00WW/NB1:8.0.0/OPR6.170623.013/00WW_4_390:user/release-keys
1517587076076 D/Debug: ro.build.version.release=8.0.0
1517587076076 D/Debug: ro.build.display.id=00WW_4_39A
1517587076076 D/Debug: ro.product.name=TA-1004_00WW
1517587076076 D/Debug: ro.product.device=NB1
1517587076076 D/Debug: ro.product.board=MSM8998
1517587076076 D/Debug: ro.product.manufacturer=HMD Global
1517587076077 D/Debug: ro.product.brand=Nokia
1517587076077 D/Debug: ro.product.model=TA-1004
1517587076077 D/Debug: ro.bootloader=unknown
```

**Comment by d4rken** [comment]
Ok, let's see.

    1517569283330 I/StorageManager: Storage(location=SDCARD, path=/storage/emulated/0, userHandle=0, flags=[PRIMARY, EMULATED], safUri=null, mount=Mountpoint(path=/storage/emulated, blockDevice=/data/media, fileSystemType=SDCARDFS, options=[noexec, rw, fsuid=1023, nodev, fsgid=1023, nosuid, noatime, multiuser, gid=9997, mask=7]))

    1517569283330 I/StorageManager: Storage(location=SDCARD, path=/storage/7F0F-0312, userHandle=-1, flags=[SECONDARY], safUri=null, mount=Mountpoint(path=/storage/7F0F-0312, blockDevice=/mnt/media_rw/7F0F-0312, fileSystemType=SDCARDFS, options=[noexec, rw, fsuid=1023, nodev, fsgid=1023, nosuid, noatime, gid=9997, mask=18]))


Storage detection looks good.

    1517569283331 D/StorageAccessFragment: WRITE_EXTERNAL_STORAGE permission has already been granted. Progressing...


Normal storage access is granted
 

    1517569283331 D/UriAccessFragment: isUriAccessNeeded(), withoutAccess=true, dontShowAgain=false


Extra access is needed.

    1517569283380 D/SetupActivity: SetupItem #0: Setup Intro

    1517569283381 D/SetupActivity: SetupItem #1: Setup StorageFramework


Setup UI screens for that are created.

    1517569284773 V/Fragment:SAFSetupFragment(214b3a): onViewCreated(view=android.widget.ScrollView{778d548 VFED.V... ......I. 0,0-0,0}, savedInstanceState=null)

    1517569284774 D/UriAccessFragment: Looking for access to [RequestObject(storage=Storage(location=SDCARD, path=/storage/7F0F-0312, userHandle=-1, flags=[SECONDARY], safUri=null, mount=Mountpoint(path=/storage/7F0F-0312, blockDevice=/mnt/media_rw/7F0F-0312, fileSystemType=SDCARDFS, options=[noexec, rw, fsuid=1023, nodev, fsgid=1023, nosuid, noatime, gid=9997, mask=18])))]


SD Maid wants access for the correct storage "/storage/7F0F-0312"

    1517569285938 V/Fragment:SAFSetupFragment(214b3a): onPause()


The pause means that the permission dialog popped into foreground.
 

    1517569287288 V/Fragment:SAFSetupFragment(214b3a): onActivityResult(requestCode=100, resultCode=-1, data=Intent { flg=0xc3 })


SD Maid gets focused again and receives the dialog result.
Requestcode - 100 = 0 which is the position that our storage has in the list, position 0.
And resultcode -1 is actually "Activity.RESULT_OK".

But then we don't get any more useful log output, the only way we can abort without anymore log data is through this:

    Uri treeUri = resultData.getData();

    if (treeUri == null) return;


The "treeUri" is the path of the location we just got granted access to, we need this URI to use in a call that basically tells the system "yep that's what we wanted, store that permission".
Hm samples show that this is the correct way to do it:
https://github.com/googlesamples/android-ScopedDirectoryAccess/blob/master/Application/src/main/java/com/example/android/scopeddirectoryaccess/ScopedDirectoryAccessFragment.java#L89

**Comment by d4rken** [comment]
When doing it manually via advanced settings it works.

    1517586551369 D/StorageVolumeMapper: Mapped VolumeRoot(treeUri=/tree/7F0F-0312:, rootId=7F0F-0312, title=SD-карта, documentId=7F0F-0312:, storagePath=/storage/7F0F-0312) to UriPermission {uri=content://com.android.externalstorage.documents/tree/7F0F-0312%3A, modeFlags=3, persistedTime=1517586551350}


SD Maid could map based on the ID `7F0F-0312` which is both part of the URI and returned by the systems storage API, so that part works great.

On the subsequent start mapping also works flawless:

    1517587083310 D/StorageVolumeMapper: Mapped VolumeRoot(treeUri=/tree/7F0F-0312:, rootId=7F0F-0312, title=SD-карта, documentId=7F0F-0312:, storagePath=/storage/7F0F-0312) to UriPermission {uri=content://com.android.externalstorage.documents/tree/7F0F-0312%3A, modeFlags=3, persistedTime=1517587036458}

    1517587083315 V/StorageVolumeMapper: getUri(): /storage/7F0F-0312/Android/media -> content://com.android.externalstorage.documents/tree/7F0F-0312%3A/document/7F0F-0312%3AAndroid%2Fmedia

    1517587083333 V/StorageVolumeMapper: getUri(): /storage/7F0F-0312 -> content://com.android.externalstorage.documents/tree/7F0F-0312%3A/document/7F0F-0312%3A

    1517587083342 V/StorageVolumeMapper: getUri(): /storage/7F0F-0312/Android/obb -> content://com.android.externalstorage.documents/tree/7F0F-0312%3A/document/7F0F-0312%3AAndroid%2Fobb

    1517587083355 V/StorageVolumeMapper: getUri(): /storage/7F0F-0312/Android/data -> content://com.android.externalstorage.documents/tree/7F0F-0312%3A/document/7F0F-0312%3AAndroid%2Fdata

**Comment by d4rken** [comment]
Looks like a Nokia bug, stock ASOP emulator shows that this should work.

**Comment by d4rken** [comment]
SD Maid will fallback to manual selection if this occurs.

**Comment by d4rken** [comment]
From affected user, fallback works:

```
1517740454575 V/Fragment:SAFSetupFragment(848cf0): onActivityResult(requestCode=100, resultCode=-1, data=Intent { flg=0xc3 })
1517740454577 E/null: Activity.RESULT_OK, but intent data is empty: Intent { flg=0xc3 }
```

**Comment by d4rken** [comment]
TL;DR 

At least this Nokia ROM is bugged such that `createAccessIntent(...)` doesn't result an `URI` in `onActivityResult`. Without the `URI` we can't call `takePersistableUriPermission` to make use of it.
